src/runtimes/docker/utils/normalizePlatform.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ContainerPlatform } from "../../../tasks/DockerBuildTaskDefinitionBase";
+import { getNativeArchitecture } from "./getNativeArchitecture";
+
+/**
+ * Normalizes a platform string or object to a standardized `ContainerPlatform` object.
+ *
+ * If a string is passed in, the string is split into `os` and `architecture` components,
+ * which are used to create a new `ContainerPlatform` object. If either component is missing,
+ * a default value of `'linux'` for `os` or the result of `getNativeArchitecture()` for
+ * `architecture` is used.
+ *
+ * If an object is passed in, the `os` and `architecture` properties are extracted and used
+ * to create a new `ContainerPlatform` object. If either property is missing, a default value
+ * of `'linux'` for `os` or the result of `getNativeArchitecture()` for `architecture` is used.
+ *
+ * @param platform The platform string or object to normalize.
+ * @returns A standardized `ContainerPlatform` object.
+ * @throws An error if the platform string is malformed.
+ */
+export function normalizePlatform(platform: string | ContainerPlatform): ContainerPlatform {
+
+    if (isPlatformEmpty(platform)) {
+        return undefined;
+    }
+    else if (typeof platform === 'string') {
+        if (!isValidString(platform)) {
+            throw new Error("Platform string is malformed. It should be in the format of '{os}/{architecture}'.");
+        }
+
+        const [os = 'linux', architecture = getNativeArchitecture()] = platform.split('/');
+        return { os, architecture };
+    }
+    else {
+        const { os = 'linux', architecture = getNativeArchitecture() } = platform;
+        return { os, architecture };
+    }
+
```

Comment:
This logic should simply normalize a string into a ContainerPlatform object without any consideration for defaults, etc. Defaults can be applied much more easily in DockerClientBase.ts when generating the --platform argument (if necessary).

```suggestion
    if (platform && typeof platform === 'string') {
        const [os, ...architectureParts] = platform.split('/');
        const architecture = architectureParts.join('/');
        
        if (!os || !architecture) {
            throw new Error('Platform string is malformed. It should be in the format "{os}/{architecture}".');
        }
        
        return { os, architecture };
    }
    
    return platform || {};
```